azext_iot/common/certops.py

```diff
@@ -122,3 +132,252 @@ def open_certificate(certificate_path: str) -> str:
             certificate = base64.b64encode(certificate).decode("utf-8")
     # Remove trailing white space from the certificate content
     return certificate.rstrip()
+
+
+def create_v3_self_signed_root_certificate(
+    subject: str = "Azure_IoT_CLI_Extension_Cert",
```

Comment:
There certainly are.

In developing the v3 extension version, since it was so specifically tailored for edge, I wanted to avoid stepping on existing certificate code until it was finalized. Now that we have it successfully validated for Edge / module certificate communication, there's definitely an opportunity to consolidate.

I'll spike on this for a bit - For the initial self-signed cert, either a way for us to pass extra properties (v3 extensions, etc) into the existing `create_self_signed_certificate` or possibly seeing if we can just apply the v3 extension configuration to existing certificates and make sure that doesn't break anything.

I'll see if that leads to any insight on how to consolidate the signing / cert chaining process for edge certs too